src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -247,7 +247,23 @@ internal ClientCertificateOption ClientCertificateOptions
             }
         }
 
-        internal X509Certificate2Collection ClientCertificates => _clientCertificates ?? (_clientCertificates = new X509Certificate2Collection());
+        internal X509Certificate2Collection ClientCertificates
+        {
+            get
+            {
+                if (_clientCertificateOption != ClientCertificateOption.Manual)
+                {
+                    throw new InvalidOperationException(SR.Format(SR.net_http_invalid_enable_first, "ClientCertificateOptions", "Manual"));
+                }
```

Comment:
I realize this is just matching netfx, but it's an odd design that you can do:
```C#
var certs = client.ClientCertificates;
client.ClientCertificateOptions = ClientCertificateOptions.Automatic;
certs.Add(...);
```
but not:
```C#
client.ClientCertificateOptions = ClientCertificateOptions.Automatic;
var certs = client.ClientCertificates;
certs.Add(...);
```

I guess it's just a best-effort attempt to help the dev, similar to how the implementation tries to stop the dev from making changes to settings after an operation has been started, but nothing prevents them from modifying, for example, ClientCertificates after that.

Reply:
> but nothing prevents them from modifying, for example, ClientCertificates after that.

From previous issue #11743 : `However, in .NET Core, the above will silently continue and ignore any certificates added to the ClientCertificates collection.` Seems the "protection" here is that, we ignore any certificates added to the ClientCerfiticates collection.

I will verify if we ignore later certificates or not.

Reply:
After some experiments, the description in original issue is not true: we actually will allow the operation and add new certificates to the collection. (at least on WinHttpHandler)

So the concern is valid, there is nothing prevents developers from modifying ClientCertificates after the initial check.

Should we open an issue to track this or document the behavior?

Reply:
>Should we open an issue to track this or document the behavior?

We can't change .NET Framework behavior.  So, if you're proposing changing .NET Core behavior to diverge from .NET Framework, then I would say, no, don't open an issue (for a product change) unless it is to document something.